pkg/schedule/affinity/manager_test.go

```diff
@@ -0,0 +1,568 @@
+// Copyright 2025 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package affinity
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/pingcap/kvproto/pkg/metapb"
+
+	"github.com/tikv/pd/pkg/core"
+	"github.com/tikv/pd/pkg/mock/mockconfig"
+	"github.com/tikv/pd/pkg/schedule/labeler"
+	"github.com/tikv/pd/pkg/storage"
+	"github.com/tikv/pd/pkg/utils/keyutil"
+)
+
+// TestGetRegionAffinityGroupState tests the GetRegionAffinityGroupState method of Manager.
+func TestGetRegionAffinityGroupState(t *testing.T) {
+	re := require.New(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	store := storage.NewStorageWithMemoryBackend()
+	storeInfos := core.NewStoresInfo()
+	for i := uint64(1); i <= 4; i++ {
+		storeInfo := core.NewStoreInfo(&metapb.Store{Id: i, Address: "test"})
+		storeInfo = storeInfo.Clone(core.SetLastHeartbeatTS(time.Now()))
+		storeInfos.PutStore(storeInfo)
+	}
+
+	conf := mockconfig.NewTestOptions()
+
+	// Create region labeler
+	regionLabeler, err := labeler.NewRegionLabeler(ctx, store, time.Second*5)
+	re.NoError(err)
+
+	manager, err := NewManager(ctx, store, storeInfos, conf, regionLabeler)
+	re.NoError(err)
+
+	// Create affinity group
+	ranges := createGroupForTest(re, manager, "test_group", 6)
+
+	// Test 0: voterStoreIDs will be sorted.
+	_, err = manager.UpdateAffinityGroupPeers("test_group", 1, []uint64{3, 2, 1})
+	re.NoError(err)
+	groupInfo := getGroupForTest(re, manager, "test_group")
+	re.True(slices.Equal([]uint64{1, 2, 3}, groupInfo.VoterStoreIDs))
+
+	// Test 1: Region not belonging to any affinity group should return false
```

Comment:
Could you modify it to the `for-testCases` format? This would make it easier to expand test cases in the future and improve maintainability. Please determine if similar modifications can be made elsewhere.